pkg/satellite/api.go

```diff
@@ -0,0 +1,121 @@
+package satellite
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/graphql-go/graphql"
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/auth"
+	"storj.io/storj/pkg/satellite/satelliteql"
+)
+
+const (
+	authorization = "Authorization"
+	contentType   = "Content-Type"
+
+	authorizationBearer = "Bearer"
+
+	applicationJSON    = "application/json"
+	applicationGraphql = "application/graphql"
+)
+
+func (gw *gateway) grapqlHandler(w http.ResponseWriter, req *http.Request) {
+	token := getToken(req)
+	//TODO: handle error
+	query, _ := getQuery(req)
+
+	result := graphql.Do(graphql.Params{
+		Schema:        gw.schema,
+		Context:       auth.WithAPIKey(context.Background(), []byte(token)),
+		RequestString: query,
+	})
+
+	var err error
+	if result.HasErrors() {
+		err = json.NewEncoder(w).Encode(result.Errors)
+	} else {
+		err = json.NewEncoder(w).Encode(result)
+	}
+
+	if err != nil {
+		gw.logger.Error(err)
+		return
+	}
+
+	gw.logger.Info(result)
+}
+
+// getToken retrieves token from request
+func getToken(req *http.Request) string {
+	value := req.Header.Get(authorization)
+	if value == "" {
+		return ""
+	}
+
+	if !strings.HasPrefix(value, authorizationBearer) {
```

Comment:
Check against `"Bearer "` rather than `"Bearer"`, because we don't want to match "BearerXYZ ".